backend/src/main/java/com/bakdata/conquery/models/datasets/concepts/filters/postalcode/PostalCodeProcessor.java

```diff
@@ -0,0 +1,64 @@
+package com.bakdata.conquery.models.datasets.concepts.filters.postalcode;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import com.univocity.parsers.common.ParsingContext;
+import com.univocity.parsers.common.processor.AbstractRowProcessor;
+import lombok.extern.slf4j.Slf4j;
+import org.glassfish.jersey.internal.guava.HashBasedTable;
+import org.glassfish.jersey.internal.guava.Table;
+
+@Slf4j
+public class PostalCodeProcessor extends AbstractRowProcessor {
+
+
+	final Table<Integer, Integer, Double> csvEntries = HashBasedTable.create();
+	private int plz1Index, plz2Index, distanceIndex;
+
+	@Override
+	public void processStarted(ParsingContext context) {
+		super.processStarted(context);
+		final String[] headers = context.headers();
+		plz1Index = IntStream.range(0, headers.length).filter(i -> headers[i].equals("plz1")).findFirst().orElse(-1);
+		if (plz1Index == -1) {
+			throw new IllegalStateException("Column plz1 not found in the csv file");
+		}
+
+		plz2Index = IntStream.range(0, headers.length).filter(i -> headers[i].equals("plz2")).findFirst().orElse(-1);
+		if (plz2Index == -1) {
+			throw new IllegalStateException("Column plz2 not found in the csv file");
+		}
+
+		distanceIndex = IntStream.range(0, headers.length).filter(i -> headers[i].equals("Distanz_convert_in_km")).findFirst().orElse(-1);
+		if (distanceIndex == -1) {
+			throw new IllegalStateException("Column Distanz_in_km not found in the csv file");
+		}
+
+
+		context.skipLines(1); //skip headers when reading rows
```

Comment:
Ich denke das kannst du dem CSV Parser direkt sagen statt hier zu skippen.

Reply:
leider wird das nur so gemacht

Reply:
Das kann ich mir nicht vorstellen, schau bitte nochmal in der Dokumentation des CSV Parser nach.

Reply:
irgendwie wird es durch die Methode `setHeaderExtractionEnabled` realisiert. Allerdings, wenn man es darüber überspringt, bekommt der Custom-Processor bei `processStarted` nicht mehr übergeben und trotzdem ist bekommt man die headers als erste Zeile in `rowProcessed` . Von daher denke ich, dass es schon Okay ist wie ich gemacht habe.